src/app/@ansyn/ansyn/modules/filters/models/counters/enum-filter-counters.ts

```diff
@@ -23,7 +23,11 @@ export class EnumFilterCounters extends FilterCounters {
 	}
 
 	incrementFilteredCount(value: any): void {
-		this.enumsFields.get(value).filteredCount = this.enumsFields.get(value).filteredCount + 1;
+		if (this.enumsFields.has(value)) {
+			this.enumsFields.get(value).filteredCount = this.enumsFields.get(value).filteredCount + 1;
+		} else {
+			this.enumsFields.set(value, { filteredCount: 1 })
```

Comment:
why do you set the filteredCount to 1 when the field's got no value?
why not 0?

Reply:
@sulimani55 Because, what this function (incrementFilteredCount) does, is to add one to the counter (because we found an overlay that matches the filter). If this field has no value, we should add counter with value 0, yes, but then we should increment it by 1. So the result is that the counter is 1.

Reply:
but if the value does not exist, what are we incrementing exactly

Reply:
We are incrementing the counter. If we got here, it means that the value does exist, but was not initialized correctly.. I hope can be understood

Reply:
+1 if it gets here and it's not initialized - i don't know if there isn't another place it should be initialized properly in 
also i don't know if this is the place to do it, and not somewhere else?

Reply:
@sulimani55 I added a comment. Does it seem clear enough?

Reply:
@Danaorlov1005 What I added is just a safely check for this particular method (incrementFilteredCount). It will prevent crash in this method, in a nice way (I think), regardless of what happens anywhere else. So even if it does not fix everything, it will improve the current situation, and will prevent the current crash.

Reply:
@Danaorlov1005 I also think we should look for a general solution for crashes in effects.

Reply:
i think the method incrementFilteredCount should only, well, increment the counter.
i dont this its wise to initiate the counter in that particular method but in another one (I suppose there's a method that initiates the filters counters - maybe look over there?)

Reply:
@ramtob I understand what you are saying, this IS a safety check and its ok
but on it's own it does not solve the problem so I agree with @sulimani55 about checking the place where the filters are initialized. 
we need to see where does the problem come from because covering for the problem here leaves other places vulnerable ..